src/main/java/com/commercetools/sync/categories/README.md

```diff
@@ -0,0 +1,221 @@
+                                                        .of(ctpClient)
+                                                        .removeOtherCollectionEntries(true)
+                                                        .build();
+  ````
+  #### `removeOtherProperties` [Optional]
+  Defines an optional field which represents a boolean flag which adds object properties (e.g. custom fields, product attributes, etc..) 
+  without deleting existing ones. If set to true, which is the default value of the option, it deletes the existing 
+  object properties. If set to false, it doesn't delete the existing ones.
+  ````java
+  // instantating options which deletes the existing object properties
+   final CategorySyncOptions categorySyncOptions = CategorySyncOptionsBuilder
+                                                        .of(ctpClient)
+                                                        .removeOtherProperties(true)
+                                                        .build();
+  ````
+  #### `updateActionsFilter` [Optional]
+  Defines an optional field which represents updateActions filter function which can be applied on generated list of 
+  update actions to produce a resultant list after the filter function has been applied.
+  ````java
+  
+  // instanting a reverse filter function
+  final Function<List<UpdateAction<Category>>, List<UpdateAction<Category>>> reverseOrderFilter = (unfilteredList) -> {
+              Collections.reverse(unfilteredList);
+              return unfilteredList;
+          };
+  // instantating options which reverses an update actions list order
+   final CategorySyncOptions categorySyncOptions = CategorySyncOptionsBuilder
+                                                        .of(ctpClient)
+                                                        .setUpdateActionsFilter(reverseOrderFilter)
+                                                        .build();
+  ````
+  
+  
+  
+
+## How does it work?
+
+The category sync uses the `externalId` to match new categories to existing ones. 
+1. If a category exists with the same `externalId`, it means that this category already exists on the CTP project. Therefore, 
+the tool calculates update actions that should be done to update the old category with the new category's fields.
+Only if there are update actions needed, they will be issued to the CTP platform.
+2. If no matching category with the same `externalId` is found, the tool will create a new one. 
+
+The sync, however, will never delete a category.
+ 
+## FAQ
+#### What does the number of processed categories actually refer to in the statistics of the sync process?
+It refers to the total number of categories which were required to be updated/created. A category that failed to be 
+updated/created will still be counted as processed. Only a category which wasn't required to be updated/created will not
+be counted as processed. 
```

Comment:
According to above description the test below should pass, shouldn't it? You could can it to `CategorySyncTest` and check:

```
@Test
    public void syncDrafts_WithExistingCategoryAndAnalogueDraft_ShouldNotIncrementCounters() {
        final ArrayList<CategoryDraft> categoryDrafts = new ArrayList<>();
        final CategoryDraft mockCategoryDraft = mock(CategoryDraft.class);
        when(mockCategoryDraft.getName()).thenReturn(LocalizedString.of(Locale.ENGLISH, "name"));
        when(mockCategoryDraft.getSlug()).thenReturn(LocalizedString.of(Locale.ENGLISH, "slug"));
        when(mockCategoryDraft.getExternalId()).thenReturn("externalId");
        when(mockCategoryDraft.getDescription()).thenReturn(LocalizedString.of(Locale.ENGLISH, "description"));
        when(mockCategoryDraft.getMetaDescription()).thenReturn(LocalizedString.of(Locale.ENGLISH,"metaDescription"));
        when(mockCategoryDraft.getMetaTitle()).thenReturn(LocalizedString.of(Locale.ENGLISH,"metaTitle"));
        when(mockCategoryDraft.getMetaKeywords()).thenReturn(LocalizedString.of(Locale.ENGLISH,"metaKeywords"));
        when(mockCategoryDraft.getOrderHint()).thenReturn("orderHint");
        when(mockCategoryDraft.getParent()).thenReturn(Reference.of("type","parentId"));
        categoryDrafts.add(mockCategoryDraft);


        categorySync.syncDrafts(categoryDrafts);
        assertThat(categorySync.getStatistics().getCreated()).isEqualTo(0);
        assertThat(categorySync.getStatistics().getFailed()).isEqualTo(0);
        assertThat(categorySync.getStatistics().getUpdated()).isEqualTo(0);
        assertThat(categorySync.getStatistics().getProcessed()).isEqualTo(0);
        assertThat(categorySync.getStatistics().getReportMessage()).isEqualTo("Summary: 0 categories were processed in total " +
                "(0 created, 0 updated and 0 categories failed to sync).");
    }
```